tlair/src/main/scala/at/forsyte/apalache/tla/lir/io/CounterexampleWriter.scala

```diff
@@ -52,59 +52,69 @@ object CounterexampleWriter {
 }
 
 class TlaCounterexampleWriter(writer: PrintWriter) extends CounterexampleWriter {
-
-  def printStateFormula(pretty: PrettyWriter, state: State) = {
+  def printStateFormula(pretty: PrettyWriter, state: State): Unit = {
     if (state.isEmpty) {
       pretty.write(ValEx(TlaBool(true)))
+      pretty.writeln()
     } else {
       state.toList.sortBy(_._1).foreach { case (name, value) =>
-        writer.print(s"/\\ $name = ")
+        pretty.write(s"/\\ $name = ")
         pretty.write(value)
-        writer.println
+        pretty.writeln()
       }
     }
   }
 
   override def write(rootModule: TlaModule, notInvariant: NotInvariant, states: List[NextState]): Unit = {
     val pretty = new PrettyWriter(writer)
 
-    writer.println("%s MODULE counterexample %s\n".format("-" * 25, "-" * 25))
-    writer.println("EXTENDS %s\n".format(rootModule.name))
+    pretty.prettyWriteDoc(pretty.moduleNameDoc("counterexample", 25))
+    pretty.writeln()
+    pretty.prettyWriteDoc(pretty.moduleExtendsDoc(rootModule.name))
+    pretty.writeln()
 
     states.zipWithIndex.foreach {
       case (state, 0) =>
-        writer.println("(* Constant initialization state *)")
-        writer.println(s"\nStateConst ==")
+        pretty.writeln("(* Constant initialization state *)")
+        pretty.writeln(s"\nConstInit ==")
```

Comment:
Why don't use use `write(decl: TlaOperDecl)`? It would do all pretty printing for you.

Reply:
Not the comments, but mostly because this was a batch 'find-replace'. Was there a reason this originally wasn't printed as a decl?

Reply:
I have no idea. Maybe because somebody wanted to write code quick %)